Sources/SwiftProtobufPluginLibrary/CodeGenerator.swift

```diff
@@ -110,10 +115,28 @@ extension CodeGenerator {
       return
     }
 
+    var extensionMap = SimpleExtensionMap()
+    if !customOptionExtensions.isEmpty {
+      for e in customOptionExtensions {
+        // Don't include Google_Protobuf_FeatureSet, that will be handing via custom features.
+        precondition(e.messageType == Google_Protobuf_EnumOptions.self ||
+                     e.messageType == Google_Protobuf_EnumValueOptions.self ||
+                     e.messageType == Google_Protobuf_ExtensionRangeOptions.self ||
+                     e.messageType == Google_Protobuf_FieldOptions.self ||
+                     e.messageType == Google_Protobuf_FileOptions.self ||
+                     e.messageType == Google_Protobuf_MethodOptions.self ||
+                     e.messageType == Google_Protobuf_OneofOptions.self ||
+                     e.messageType == Google_Protobuf_ServiceOptions.self,
+                     "CodeGenerator `customOptionExtensions` must only extend the descriptor.proto 'Options' messages \(e.messageType).")
```

Comment:
Shouldn't we also allow `Google_Protobuf_MessageOptions`?

Reply:
Oops, yup.